daemon/endpoint.go

```diff
@@ -385,14 +384,6 @@ func NewPatchEndpointIDHandler(d *Daemon) PatchEndpointIDHandler {
 	return &patchEndpointID{d: d}
 }
 
-func validPatchTransitionState(state models.EndpointState) bool {
```

Comment:
I suppose the reason this was a standalone function was that PATCH was a specific operation outside of the endpoint. It might be more clear to rename this to something that reflects what we are really checking within the endpoint.

Reply:
Looking at the way we use this some more, I'm not sure what we gain by moving this function into pkg/endpoint. It really is specific to API itself and whether it allows something. If it were part of the endpoint then it could be part of ProcessChangeRequest (which might be better since the endpoint is then doing the check not the caller, and it already does it once).

Reply:
This is a symptom of the problem that we rely on endpoint's being on specific state in order to perform specific operations on them. I think this PR makes that problem more acute. I think the problem is that the daemon is aware of the endpoint's state at all. One of my goals here is to try to encapsulate the information about the endpoint away from callers, which includes its state. I am going to merge with that in mind, as this is more of an interim "let's make things better than they previously were" PR, which still has some weaknesses.

Reply:
Hm, I may be able to do this actually...

Reply:
Done!